bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/storage/Storage.java

```diff
@@ -2144,7 +2144,7 @@ public String copyToTempLibrary(Generation generation, String absolutePath) {
 		// in cases where the temp dir may already exist.
 		Long bundleID = Long.valueOf(generation.getBundleInfo().getBundleId());
 		for (int i = 0; i < Integer.MAX_VALUE; i++) {
-			bundleTempDir = new File(libTempDir, bundleID.toString() + "_" + Integer.valueOf(i).toString()); //$NON-NLS-1$
+			bundleTempDir = new File(libTempDir, bundleID.toString() + "_" + Integer.toString(i)); //$NON-NLS-1$
```

Comment:
Not that this matters much for this code block since it is expected that we will iterate very few times before finding an available directory name (mostly just once), but this change avoids the Integer.valueOf optimization for the small Integer cache.

If this is about simplifying the code then it would be more simple to avoid any method calls and just `+ i` onto the string concatination:

`bundleID.toString() + "_" + i`